src/main/kotlin/com/strumenta/kolasu/model/Processing.kt

```diff
@@ -272,12 +272,45 @@ fun Node.mapTree(operation: (Node) -> Node): Node {
 }
 
 /**
- * Replace this node with "other" (by modifying the children of the parent node.)
- * For this to work, assignParents() must have been called.
+ * Replace [this] node with [other] (by modifying the children of the parent node.)
+ * For this to work, [Node.assignParents] must have been called.
  */
 fun Node.replaceWith(other: Node) {
     if (this.parent == null) {
         throw IllegalStateException("Parent not set")
     }
     this.parent!!.transformTree { if (it == this) other else it }
 }
+
+/**
+ * Looks for [oldNode] in the lists of nodes in this node.
+ * When found, it is removed, and in its place the [newNodes] are inserted.
+ */
+@Suppress("UNCHECKED_CAST") // assumption: a MutableList with a Node in it is a MutableList<Node>
+fun Node.replaceWithSeveral(oldNode: Node, newNodes: List<Node>) {
+    relevantMemberProperties().forEach { property ->
+        when (val value = property.get(this)) {
+            is MutableList<*> -> {
```

Comment:
what if the value is the single node we were looking for? Should we throw an exception?

Reply:
Oh, we're asking each other the same thing :-)

Reply:
For my use case it's an error, so I'll make it an error.